header.c

```diff
@@ -1642,6 +1642,7 @@ int sam_hdr_remove_except(sam_hdr_t *bh, const char *type, const char *ID_key, c
     return 0;
 }
 
+// cppcheck-suppress funcArgOrderDifferent
 int sam_hdr_remove_lines(sam_hdr_t *bh, const char *type, const char *id, void *h) {
```

Comment:
Could be fixed instead by renaming the last parameter:

```suggestion
int sam_hdr_remove_lines(sam_hdr_t *bh, const char *type, const char *id, void *vrh) {
```